codegen/smithy-ruby-codegen-test/integration-specs/compression_spec.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module WhiteLabel
+  describe Config do
+    context 'disable_request_compression' do
+      it 'raises error when given an invalid input' do
+        expect { Config.new(disable_request_compression: 'string') }
+          .to raise_error(
+            ArgumentError,
+            'Expected config[:disable_request_compression] to be in [TrueClass, FalseClass], got String.'
+          )
+      end
+    end
+
+    context 'request_min_compression_size_bytes' do
+      it 'raises error when given invalid integer' do
+        # TODO after we implement constraints on configs
```

Comment:
This should be merged now

Reply:
I dont think we ever added the ability to do arbitrary constraints, such as enforcing the range of a number.

Reply:
My current plan is as follows:
- merge this PR now
- make a separate PR to address the constraints
- another PR for updating the middleware relative ordering
- a quick PR to wrap up testing

I would like to keep the PRs separately. They're all about different features and it'd probably be easier to review and adding each one independently.